### ISSUE TITLE
what is the difference between tensorrt and inference server and tf serving?

### ISSUE THREAD

**Issue by merlinzone** [NVIDIA/dl-inference-server#4]
what is the difference between tensorrt and inference server?

what is the difference between tensorflow serving and  inference server?

what is the development roadmap for inference server?

**Comment by deadeyegoodwin** [comment]
"NVIDIA TensorRT is a high-performance deep learning inference optimizer and runtime..."
See https://developer.nvidia.com/tensorrt. TensorRT is delivered as a shared library (with C++ and python bindings) that you can use to construct a network and then optimize it for inference on NV GPUs. Typically users import a trained network from TensorFlow or ONNX and use TensorRT to optimize it for inference. TensorFlow recently incorporated TensorRT as an internal optimizer.

NV Inference Server is conceptually similar to TFServing in that it is an  "inferencing service" that you can run in the cloud/datacenter/workstation to provide a service endpoint (HTTP REST currently). It simplifies inferencing in these environments because a client can perform inferencing by simply sending a request to the server with appropriate inputs.

Inference Server uses TensorRT runtime to serve models (called TensorRT PLANs) generated by TensorRT. Inference server can also serve TensorFlow models and in the future will serve other model types as well (e.g. ONNX). Inference Server is focused on high-performance, GPU accelerated inferencing that takes full advantage of NVIDIA technologies like TensorRT while at the same time supporting a wide range of model types.

We are considering a wide range of future development options and will release updates at a monthly cadence. Are their specific features you are interested in? We would like to hear your feedback.

**Comment by AbdallahNasir** [comment]
@deadeyegoodwin Suppose I have a frozen model ready to be deployed on Tensorflow serving. What do I need to deploy it on Inference Server optimized by TensorRT?

**Comment by deadeyegoodwin** [comment]
You need to set up a model repository. Your need the frozen model file itself and a config.pbtxt file describing the inputs/outputs and some other (optional) information. The process is described here: https://docs.nvidia.com/deeplearning/sdk/inference-user-guide/index.html#model

Let us know if you are able to follow those instructions or if you still need additional guidance.

**Comment by AbdallahNasir** [comment]
Thanks for the response. 

So I do not need to convert the Tensorflow pb frozen model to UFF to benefit from TensorRT optimizations? 

This also applies to models from the tensorflow object detection API? like Faster R-CNN Resnet 101 COCO?
http://download.tensorflow.org/models/object_detection/faster_rcnn_resnet101_coco_2018_01_28.tar.gz
from
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md

The input to these models got resized by the model, this snippet is from pipeline.config:
    image_resizer {
      keep_aspect_ratio_resizer {
        min_dimension: 600
        max_dimension: 1024
      }
    }
Can I do something similar in the config.pbtxt?

Thanks dear @deadeyegoodwin ,
